XRTK.SDK/Packages/com.xrtk.sdk/Features/UX/Scripts/Controllers/Hands/BaseHandVisualizer.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using XRTK.Definitions.HandTracking;
+using XRTK.Definitions.Utilities;
+using XRTK.EventDatum.Input;
+using XRTK.Interfaces.InputSystem.Handlers;
+using XRTK.Providers.Controllers.Hands;
+using XRTK.Services;
+
+namespace XRTK.SDK.UX.Controllers.Hands
+{
+    public class BaseHandVisualizer : DefaultMixedRealityControllerVisualizer, IMixedRealitySourceStateHandler, IMixedRealityHandJointHandler, IMixedRealityHandMeshHandler
```

Comment:
Rather than base of the "Default", since the GameObjectProxy is not used here, should this not inherit from ControllerPoseSynchronizer and IMixedRealityControllerVisualiser - as an alternate implementation?

Reply:
What do you mean not used? I guess it might be useful at some point to have it. `ControllerPoseSynchronizer` is just a MonoBehaviour as well so it has a gameObject too. The Proxy seems just a shortcut. I am also nut sure about the `ControllerFinder` tool. Might be useful to use it to find hands.

Reply:
Just noticed
```
public GameObject GameObjectProxy
        {
            get
            {
                try
                {
                    return gameObject;
                }
                catch (Exception)
                {
                    return null;
                }
            }
        }
```

Am I missing something here? How's this going to throw an exception? Can't this be simpliefied to `GameObjectProxy => gameObject`?

Reply:
Throws an exception if the native object is destroyed